indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/InputSourceSplitParallelIndexTaskRunner.java

```diff
@@ -86,13 +83,8 @@ final SubTaskSpec<T> newTaskSpec(InputSplit split)
   {
     final FirehoseFactory firehoseFactory;
     final InputSource inputSource;
-    if (baseInputSource instanceof FirehoseFactoryToInputSourceAdaptor) {
-      firehoseFactory = ((FirehoseFactoryToInputSourceAdaptor) baseInputSource).getFirehoseFactory().withSplit(split);
-      inputSource = null;
-    } else {
-      firehoseFactory = null;
-      inputSource = baseInputSource.withSplit(split);
-    }
+    firehoseFactory = null;
+    inputSource = baseInputSource.withSplit(split);
```

Comment:
would this work if baseInputSource is an adapter input source from a parser?

Reply:
As mentioned in the other comment `baseInputSource` can't be an instance of adapter input source since it's generated by `IndexTask.IndexIOConfig.getNonNullInputSource()` from `InputRowParser` & `FiniteFirehoseFactory` previously in case `InputSource` is null, now that `FiniteFirehoseFactory` is removed user has to submit non-null input source for batch index tasks for all the cases where `FiniteFirehoseFactory` instantiating spec was used.